src/pages_/_app.tsx

```diff
@@ -2,19 +2,22 @@ import React from 'react';
 import { DefaultSeo } from 'next-seo';
 import { ThemeProvider } from 'styled-components';
 import { Provider } from 'react-redux';
+import { PersistGate } from 'redux-persist/integration/react';
 import { theme } from '../utils/styles';
 import { createSEOConfig } from '../utils/seo';
 import Footer from '../components/Footer';
-import store from '../redux/store';
+import store, { persistor } from '../redux/store';
 
 function MyApp({ Component, pageProps }): JSX.Element {
   return (
     <Provider store={store}>
-      <ThemeProvider theme={theme}>
-        <DefaultSeo {...createSEOConfig({})} />
-        <Component {...pageProps} />
-        <Footer />
-      </ThemeProvider>
+      <PersistGate loading={null} persistor={persistor}>
```

Comment:
The loading time is too small imo to add a loading component, it's helpful for apps that store their full redux store in localstorage. We can reassess and change it as our app gets bigger.

Reply:
Is the loading time negligible across all devices? If there are some low-powered devices that might experience a noticeable delay we could get the best of both worlds by having a “delayed spinner” that will only kick in after a timeout. In that case, we avoid flashing a loading spinner on devices where the delay is minimal and show indications to the user that something is happening on lower-powered devices that take a little longer.

Not something to block this particular PR on - just a drive-by thought!

Reply:
1) Yeah, the loading time is niglible on all devices, if there's no localstorage data, the `PersistGate` component loads the application right away. So there's no observable difference for first-time users and web crawlers. For returning users, the storage payload is pretty small, and the data is read locally, I'm suspecting that it'll be neglible too. 

2) I'm open to the idea of having a loading component that appears after 500ms or so, we should do a performance rundown once we have a working version of the app iA and identify the bottlenecks :)

Reply:
Hmm. after looking at the code, this will show empty on SSR, no?

Reply:
Good catch @mmahalwy. Created `Provider.tsx` to fix this issue. 

We now only load `<PersistGate />` on the client. I confirmed that the DOM is loaded for SSG, and that redux-persist still works as expected